scripts/irods/test/test_pam_password.py

```diff
@@ -0,0 +1,225 @@
+            "no_extend": "",
+            "password_length": "",
+            "password_max_time": "",
+            "password_min_time": ""
+        }
+
+        _, out, _ = self.admin.assert_icommand(['iadmin', 'get_pam_password_config'], 'STDOUT', 'password_length')
+        self.original_configs = json.loads(out)
+
+        # Set the pam_password configuration values to the defaults.
+        for option_name in self.default_setup:
+            self.admin.assert_icommand(
+                ['iadmin', 'set_pam_password_config', option_name, self.default_setup[option_name]])
+
+    @classmethod
+    def tearDownClass(self):
+        # Put everything back to the original configuration settings.
+        for option_name in self.original_configs:
+            self.admin.assert_icommand(
+                ['iadmin', 'set_pam_password_config', option_name, self.original_configs[option_name]])
+
+        with session.make_session_for_existing_admin() as admin_session:
+            self.admin.__exit__()
+            admin_session.assert_icommand(['iadmin', 'rmuser', self.admin.username])
+
+    def test_no_parameters(self):
+        self.admin.assert_icommand(
+            ['iadmin', 'set_pam_password_config'], 'STDERR', 'Invalid pam_password configuration option')
+
+        # Assert that nothing changed due to the error.
+        expected_output = self.default_setup
+        self.admin.assert_icommand(
+            ['iadmin', 'get_pam_password_config'], 'STDOUT', json.dumps(expected_output, indent=4))
+
+    def test_invalid_option_name(self):
+        self.admin.assert_icommand(
+            ['iadmin', 'set_pam_password_config', 'invalid'], 'STDERR', 'Invalid pam_password configuration option')
+
+        # Assert that nothing changed due to the error.
+        expected_output = self.default_setup
+        self.admin.assert_icommand(
+            ['iadmin', 'get_pam_password_config'], 'STDOUT', json.dumps(expected_output, indent=4))
+
+    def test_empty_value(self):
+        option_name = 'password_length'
+        option_value = 500
+        self.admin.assert_icommand(['iadmin', 'set_pam_password_config', option_name, str(option_value)])
+
+        # Assert that the value has been updated to the expected new value.
+        expected_output = self.default_setup.copy()
+        expected_output['password_length'] = str(option_value)
+        self.admin.assert_icommand(
+            ['iadmin', 'get_pam_password_config'], 'STDOUT', f'"{option_name}": "{option_value}"')
+
+        # Assert that the value has been set to an empty string.
+        self.admin.assert_icommand(['iadmin', 'set_pam_password_config', option_name])
+        self.admin.assert_icommand(
+            ['iadmin', 'get_pam_password_config'], 'STDOUT', f'"{option_name}": ""')
+
+    def test_really_long_option_value(self):
+        # The input buffer to set_grid_configuration_value API is only 2700 characters long.
+        really_long_option_value = 'this_is_27_characters_long_' * 100
```

Comment:
so sending exactly 2700 is the problem?   because 2700 plus the null terminator is 2701?

i think a slightly better comment will make this clear

Reply:
Right, if you send 2700 characters, packstruct falls over; which is good, but in automated testing, the manner of falling over is interesting... can discuss. I'll update this comment.